### PR TITLE
Add version check r non noarch

### PR DESCRIPTION
**Pull request by npavlovikj**
Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Comment by npavlovikj** [comment]
I used 2 `if` checks in case the recipe has both `python` and `r-base` dependencies. If I should use both `python` and `r-base` checks in one condition instead, please let me know.

**Comment by isuruf** [comment]
Thanks @npavlovikj for the PR.
Is it possible to do this in a loop for `python` and `r-base` instead of copy-pasting the code.

**Comment by npavlovikj** [comment]
Thanks for the feedback @isuruf  - I wasn't sure how much I can modify the initial code...

**Comment by scopatz** [comment]
This looks good to me! Thanks @npavlovikj!

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -453,33 +453,41 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
...
+            for reqs in [filtered_host_reqs, filtered_run_reqs]:
+                if str(language) in reqs:
+                    continue
+                for req in reqs:
+                    constraint = req.split(" ", 1)[1]
+                    if constraint.startswith(">") or constraint.startswith(
+                        "<"
+                    ):
+                        lints.append(
+                            "Non noarch: {0} packages should have a {0} requirement without any version constraints.".format(
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py]:
```suggestion
                            "Non noarch packages should have a {0} requirement without any version constraints.".format(
```

**Comment by npavlovikj** [comment]
@isuruf , I should also add test for `r-base` in "test_lint_recipe.py", right?
Something like:
```
def test_r-base_requirements(self):
    meta = {"requirements": {"host": ["r-base >=3.5"]}}
    lints, hints = linter.lintify(meta)
    self.assertIn(
        "If r-base is a host requirement, it should be a run requirement.",
        lints,
    )
...
```
?

If I do add this, do I call `def test_r-base_requirements` elsewhere? Beside the definition, I couldn't find anything related in the repo.

**Comment by npavlovikj** [comment]
@isuruf , any comments on this?

**Comment by isuruf** [comment]
Yes, that's fine as a test. Name needs to be `test_r_base_requirements` (i.e. no dashes) and pytest will discover it automatically.

**Comment by npavlovikj** [comment]
Thank you @isuruf , I updated the PR accordingly.

**Comment by scopatz** [comment]
Thanks @npavlovikj!

**Comment by npavlovikj** [comment]
Thank you for the review and suggestions @scopatz and @isuruf !

**Pull request merged**